File: server/src/models/Session.js
/**
 * Pseudocode:
 * 1) Document Session model shape.
 * 2) Attach summary stats later.
 */

/**
 * @typedef {Object} Session
 * @property {string} sessionId
 * @property {string} userId
 * @property {string} configId
 * @property {string} startTime
 * @property {string=} endTime
 * @property {"completed"|"quit"|"failed"=} outcome
 * @property {number=} totalTimeSeconds
 * @property {number=} totalFails
 * @property {number=} totalTokensSpent
 */

export const SESSION_MODEL_DOCS = {};

File: server/src/models/Session.java
package models;

/**
 * Pseudocode:
 * 1) Document Session model shape.
 * 2) Attach summary stats later.
 */
public class Session {
  public String sessionId;
  public String userId;
  public String configId;
  public String startTime;
  public String endTime;
  public String outcome;
  public Integer totalTimeSeconds;
  public Integer totalFails;
  public Integer totalTokensSpent;
}
